lib/common/exceptions/external/sentry_ignored_gateway_timeout.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module Common
+  module Exceptions
+    class SentryIgnoredGatewayTimeout < GatewayTimeout; end
```

Comment:
This `SentryIgnoredGatewayTimeout` exception
- Inherits from `GatewayTimeout`
- added to Sentry `excluded_exceptions`
- added to the `SKIP_SENTRY_EXCEPTION_TYPES` array 

results in

- we don't have to `rescue` and `raise` in jobs 
- `ApplicationController`'s `rescue_from 'Exception'` block still treats this like a `GatewayTimeout`, generating the expected response